sympy/core/mul.py

```diff
@@ -177,7 +177,28 @@ def could_extract_minus_sign(self):
             return False  # e.g. zoo*x == -zoo*x
         c = self.args[0]
         return c.is_Number and c.is_extended_negative
+    def convert_to(self, other, unit_system="SI"):
+        """
+        Convert the quantity to another quantity of same dimensions.
+
+
+        Examples
+        ========
 
+
+        >>> from sympy.physics.units import speed_of_light, meter, second
+        >>> speed_of_light
+        speed_of_light
+        >>> (3*speed_of_light).convert_to(meter/second)
+        899377374*meter/second
+
+
+        >>> from sympy.physics.units import liter
+        >>> (26*liter).convert_to(meter**3)
+        13*meter**3/500
+        """
+        from sympy.physics.units.util import convert_to
+        return convert_to(self, other, unit_system)
```

Comment:
This circular dependency throws a red flag for me. Are there other unit codes in cor/mul.py? If not, I doubt it is a good idea to add this code in the core. I don't know how the units package works, but hopefully there is a way to make this work by only adding code in that package.

Reply:
Yeah I see what you're saying, but I don't have a good idea of how I would do it outside of the mul class, since a constant times a unit is a mul object, so I think you would have to make a change to the mul class.  can you think of any ideas? If not this functionality is probably not needed, seeing as the same thing can be achieved just by calling the static convert_to.

Reply:
I don't have any ideas, because I'm not familiar with the units code. You could ask on the mailing list.

Reply:
We should make a function rather than a method.

Reply:
We already have the function `convert_to`. Here are ways of doing this already:
```python
>>> from sympy.physics.units import *

# use the function

>>> convert_to(2*ft,inches)
24*inch

# use replace

>>> (2*ft).replace(lambda x:isinstance(x,Unit),lambda x: x.convert_to(inches))
24*inch

# make your own convenience function

>>> Mul.c2 = lambda self,u:self.replace(lambda x:isinstance(x,Unit),lambda x: x.convert_to(u))
>>> (2*ft).c2(inches)
24*inch
```

I'm not sure how complex things can get, though: we have numbers, unit and uncertainties. What is being added is a mathod for handling the unit aspect of a measurement.

Reply:
I think that this doesn't need any fixes. We may close the issue.

Reply:
> This change was made to address a TODO line in
sympy.physics.units which requested
mul objects to have a convert_to method.

Anyway, I need to check the reason for this motivation. Where did the original `TODO` line came from?
I can't see in https://github.com/sympy/sympy/blob/master/sympy/physics/units/__init__.py

Reply:
it's from line 52/53 in https://github.com/sympy/sympy/blob/master/sympy/physics/units/tests/test_quantities.py

    # TODO: eventually support this kind of conversion:
    # assert (2*speed_of_light).convert_to(m / s) == 2 * 299792458 * m / s

Reply:
I'm not sure there is good reason for the change either, I just assumed this line was put there for some reason.

Reply:
I think that you'd remove such miscellaneous comment there
Maybe @Upabjojr have made such comments in https://github.com/sympy/sympy/pull/12246, or I'm not sure about it because it was just a refactory PR.

> I'm not sure how complex things can get, though: we have numbers, unit and uncertainties. What is being added is a mathod for handling the unit aspect of a measurement.

Anyway, even though we were eventually going to fix in this direction, I we may have discussed fix like:

- `Add` should have `convert_to`
- `Pow` should have `convert_to`
- ...

or it will be incomplete. so it's very complicated.